daphne/src/roles/leader.rs

```diff
@@ -385,59 +392,67 @@ pub trait DapLeader<S>: DapAuthorizedSender<S> + DapAggregator<S> {
+                            task_config.version,
+                        ),
+                        resource: agg_job_id.for_request_path(),
+                        req_data: agg_job_cont_req.get_encoded_with_param(&task_config.version),
+                        method: LeaderHttpRequestMethod::Post,
+                    },
+                )
+                .await?;
+                let agg_job_resp = AggregationJobResp::get_decoded(&resp.payload)
+                    .map_err(|e| DapAbort::from_codec_error(e, task_id.clone()))?;
+
+                // Handle AggregationJobResp.
+                task_config.vdaf.handle_final_agg_job_resp(
+                    task_config,
+                    uncommited,
+                    agg_job_resp,
+                    metrics,
+                )?
             }
-            DapLeaderTransition::Skip => return Ok(0),
-            DapLeaderTransition::Continue(..) => {
+            DapLeaderAggregationJobTransition::Finished(agg_span) => {
+                if agg_span.report_count() > 0 {
+                    agg_span
+                } else {
+                    return Ok(0);
+                }
+            }
+            DapLeaderAggregationJobTransition::Continued(..) => {
                 return Err(fatal_error!(err = "unexpected state transition (continue)").into())
             }
         };
 
-        // Send AggregationJobContinueReq and receive AggregationJobResp.
-        let resp = leader_send_http_request(
-            self,
-            task_id,
-            task_config,
-            LeaderHttpRequestOptions {
-                path: &url_path,
-                req_media_type: DapMediaType::AggregationJobContinueReq,
-                resp_media_type: DapMediaType::agg_job_cont_resp_for_version(task_config.version),
-                resource: agg_job_id.for_request_path(),
-                req_data: agg_job_cont_req.get_encoded_with_param(&task_config.version),
-                method: LeaderHttpRequestMethod::Post,
-            },
-        )
-        .await?;
-        let agg_job_resp = AggregationJobResp::get_decoded(&resp.payload)
-            .map_err(|e| DapAbort::from_codec_error(e, task_id.clone()))?;
-
-        // Commit the output shares.
-        let agg_share_span = task_config.vdaf.handle_final_agg_job_resp(
-            task_config,
-            uncommited,
-            agg_job_resp,
-            metrics,
-        )?;
-        let out_shares_count = agg_share_span.report_count() as u64;
+        let out_shares_count = agg_span.report_count() as u64;
 
         // At this point we're committed to aggregating the reports: if we do detect a report was
         // replayed at this stage, then we may end up with a batch mismatch. However, this should
         // only happen if there are multiple aggregation jobs in-flight that include the same
         // report.
 
+        // TODO(cjaptton) Treat a failure here as fatal.
```

Comment:
Let me know if you agree and I'll file an issue so we don't forget.

Reply:
I think we should either make it fatal now (in this PR) or not at all, opening an issue pointing to a part of the code that is incomplete and not used in prod (leader) I think is counterproductive. It's not something we need to fix yet and it's likely something we don't know how to fix yet. It's just gonna "rot away" in the issues tab

Reply:
Done!